intelmq/bots/parsers/fireeye/parser.py

```diff
@@ -0,0 +1,92 @@
+
+"""
+Fireeye Parser Bot
+Retrieves a base64 encoded JSON-String from raw and converts it into an
+event.
+"""
+from intelmq.lib import utils
+from intelmq.lib.bot import ParserBot
+from intelmq.lib.message import MessageFactory
+from intelmq.lib.utils import base64_decode
+import json
+import xmltodict
+import ipaddress
+
+class FireeyeParserBot(ParserBot):
+
+
+    def process(self):
+        report = self.receive_message()
+        raw_report = utils.base64_decode(report.get('raw'))
+        my_dict = xmltodict.parse(raw_report)
+        try:
+            event = self.new_event(report)
+            for indicator in my_dict['OpenIOC']['criteria']['Indicator']['IndicatorItem']:
+                hashValue = indicator['Content']['#text']
+                indicatorType = indicator['Context']['@search']
+                if indicatorType == 'FileItem/Md5sum':
+                    event.add('malware.hash.md5', indicator['Content']['#text'])
+                if indicatorType == 'FileItem/Sha256sum':
+                    self.logger.debug('FileItem/Md5sum aus uuid' + indicator['Content']['#text'])
+                    event.add('malware.hash.sha256', indicator['Content']['#text'])
+                    self.send_message(event)
+                    data = raw_report.split('<Indicator id')
+                    uuidres = data[0].split('"alert_id">')
+                    uuid = uuidres[1].split('"')
+                    self.logger.debug("My UUDI is:  " + uuid[0])
+                    data.pop(0)
+                    data.pop(0)
```

Comment:
The parsing looks interesting. Could you please share an anonymized example as part of the tests? Then we can have a look if the parsing can be simplified and made more robust.